### PR TITLE
Feature rancher integration

### PR DESCRIPTION
**Pull request by SantoDE**
New Provider - Rancher. I still need to do some stuff, but it's a base for discussions :)

TODO:
 - [x] write tests
 - [ ] update docs
 - [X] make sure provider.Watch is implemented correctly
 - [X] set defaults

### CODE REVIEW

File: provider/rancher.go [hunk 1]
```diff
@@ -0,0 +1,458 @@
+package provider
+
+import (
```
Comment (vdemeester) [inline provider/rancher.go]: Import are not in the right order 😅

File: provider/rancher.go [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+	Health     string
+}
+
+func (r rancherData) String() string {
+	return fmt.Sprintf("{name:%s, labels:%v, containers: %v, health: %s}", r.Name, r.Labels, r.Containers, r.Health)
+}
+
+// Frontend Labels
+func (provider *Rancher) getPassHostHeader(service rancherData) string {
+	if passHostHeader, err := getServiceLabel(service, "traefik.frontend.passHostHeader"); err == nil {
```
Comment (vdemeester) [inline provider/rancher.go]: I really feel this kind of line are duplicated a lot, it's orthogonal to this PR (/me thinking outloud 🤔) but at some point I think we could refactor this to share some common code.

**Comment by emilevauge** [comment]
@SantoDE I will merge this one as is: could you open an issue to update doc & imports?

**Comment by SantoDE** [comment]
Done @emilevauge

**Comment by ashmckenzie** [comment]
Really keen to try this out, any chance of an example config?

**Comment by SantoDE** [comment]
Hey @ashmckenzie,

until PR #1127 is merged, the only way to test that feature is by providing the endpoint, access and secret key. 

You can find documentation in that PR [here](https://github.com/containous/traefik/pull/1127/files#diff-aa69786b55fa6490cf44b622e3576b1b)

**Comment by kelchm** [comment]
For anyone who is looking for it, here is a [link ](https://github.com/containous/traefik/blob/v1.2/docs/toml.md#rancher-backend)to the rancher backend provider docs.

**Pull request merged**